### PR TITLE
bump golangci to v0.38.0

### PR DESCRIPTION
**Pull request by mistikel**
**What type of PR is this? **
/kind other

**What this PR does / why we need it**:
currently, golangci is updated to v0.38.0 and `maligned` is deprecated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1259

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mistikel** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

CLA Signed

File: .golangci.yml [hunk 1]
```diff
@@ -21,7 +21,6 @@ linters:
     - govet
     - ineffassign
     - interfacer
-    - maligned
```
Comment (devigned) [inline .golangci.yml:24]: Why was this removed?

Comment (CecileRobertMichon) [reply .golangci.yml:24]: maligned was deprecated (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1259)

**Review by devigned — CHANGES_REQUESTED** [review]
```
azure/services/bastionhosts/bastionhosts.go:78:3: ineffectual assignment to publicIP (ineffassign)
		publicIP := network.PublicIPAddress{}
		^
```

Looks like the linter picked up something new. Please fix the new linter error.

**Review by devigned — APPROVED** [review]
/lgtm
/approve

Thank you for the PR!
